### PR TITLE
Middleware for new REST API responses - Subtask #225 - Closes #857

### LINKED ISSUE #857: Add new standard middleware for REST API
The old method was not very good for handling generic responses in code. We should build a middleware for the new API to use to make adding/changing the API easy.

### PR DESCRIPTION
**Pull request by MaciejBaj**
Parent issue: #225 
Child issue: #857
Lay out the foundations for REST API messages including status codes. 
In case of API failure, `ApiError` instance should be returned. HTTP middleware will handle error messages return in expected format: `{code: 500, message: 'Error message'}`.
In case of success, but empty result array code 204 is being returned.

### CODE REVIEW

File: test/unit/helpers/apiError.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+var express = require('express');
+var ApiError = require('../../../helpers/apiError.js');
+
+describe('helpers/apiError', function () {
+
+	var apiError;
+	var validErrorMessage;
+	var validErrorCode;
+
+	beforeEach(function () {
```
Comment (4miners) [inline test/unit/helpers/apiError.js]: just `before` sould be ok

Comment (MaciejBaj) [reply test/unit/helpers/apiError.js]: #812 - 1 c)

Comment (4miners) [reply test/unit/helpers/apiError.js]: For me issue you referenced looks like proposal, has this been accepted as a standards?

Comment (MaciejBaj) [reply test/unit/helpers/apiError.js]: At least it is for me. Feel free to propose another solution if you don't agree.

File: helpers/httpApi.js [hunk 2]
```diff
@@ -192,6 +194,26 @@ function respond (res, err, response) {
...
+ */
+function respondWithCode (res, err, response) {
+	if (err) {
+		res.code(err.code || apiCodes.ERROR_DEFAULT).json(err.toJson());
+	} else {
+		var isResponseEmpty = function (response) {
+			var firstValue = _(response).values().first();
+			return _.isArray(firstValue) && _.isEmpty(firstValue);
+		};
+		return res.code(isResponseEmpty(response) ? apiCodes.EMPTY_RESOURCES_SUCCESS : apiCodes.SUCCESS).json(response);
```
Comment (SargeKhan) [inline helpers/httpApi.js]: In case of non-empty resource, we only return status code `api.Codes.SUCCESS` and no other status code?

**Pull request merged**